lib/ansible/galaxy/dependency_resolution/providers.py

```diff
@@ -102,86 +104,7 @@ def identify(self, requirement_or_candidate):
         """
         return requirement_or_candidate.canonical_package_id
 
-    def get_preference(self, *args, **kwargs):
-        # type: (t.Any, t.Any) -> t.Union[float, int]
```

Comment:
Why have several large doctstrings been removed and then duplicated in several other locations?